themes/default/content/blog/next-level-iac-package-ecosystems/index.md

```diff
@@ -60,26 +64,95 @@ export const bucketName = bucket.id;
 
 // Export the URL of the file in the S3 bucket
 export const objectUrl = pulumi.interpolate`s3://${bucket.id}/${bucketObject.key}`;
-``` 
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import pulumi
+import pulumi_aws as aws
+
+# Create an S3 bucket
+bucket = aws.s3.Bucket('my-bucket')
+
+# The path to the file to upload to S3
+file_path = "path/to/your/local/file.ext"  # Replace with the path to the file you want to upload
+
+# Read the contents of the file to be uploaded to S3.
+with open(file_path, 'rb') as file:
+    file_contents = file.read()
+
+# Upload the file to the S3 bucket
+bucket_object = aws.s3.BucketObject('my-bucket-object',
+                                bucket=bucket.id, # Reference to the bucket created above.
+                                key='file.ext',  # This will be the file's name in the bucket.
+                                content=file_contents,
+                                content_type='text/plain')  # MIME-type of the file. Adjust if you upload a different type of file.
+
+# Export the name of the bucket
+pulumi.export('bucket_name', bucket.id)
+
+# Export the URL of the file in the S3 bucket
+pulumi.export('object_url', pulumi.Output.concat('s3://', bucket.id, '/', bucket_object.key))
+```
+
+{{% /choosable %}}
+
 
 However, something that stands out as a weak link here is the line:
 
+{{< chooser language "typescript,python" />}}
+
+{{% choosable language typescript %}}
+
 ```typescript
 contentType: "text/plain", // MIME-type of the file. Adjust if you upload a different type of file.
 ```
 
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+content_type='text/plain')  # MIME-type of the file. Adjust if you upload a different type of file.
+```
+{{% /choosable %}}
+
 Yeah, hard-coding the MIME type for every file I need to upload is a not going to scale. In fact, I’m probably going to get a random directory full of files I need to iterate over, each with different file types, and it will change every time it runs, and we won’t know what type of file it is ahead of time.
 
-Packages to the rescue!! Turns out that determining the MIME type of a file in TypeScript only involves importing the community-created [`mime-types`][npm-mimetypes] library and a single function call to `mime.lookup(...)`:
+Packages to the rescue!! Turns out that determining the MIME type of a file in TypeScript only involves importing a community-created [`mime-types`][npm-mimetypes] [library][python-mimetypes] and a single function call to guess the mimetype:
```

Comment:
This confused me a bit -- partly because it's two links that look like one, but also because in the case of Python, it isn't a community-created library, but part of (I think) the Python standard library. So we may way to choose an actual package instead, to illustrate the point of the post a bit more clearly. (Even if some users may realize this is built into Python itself.)

Reply:
Well, one of the issues was that I wrote this to be Typescript specific, then I was asked to add Python in today. This made it a little complicated because the example works for Typescript, but not as well as for Python. This is one of the issues when we talk about package ecosystems -- they are not homogenous and what is simple and easy for one is not going to be the same in the next one, so multilingual examples will end up being more complicated to show. That said, even though the Python library is part of the core library, it is still a community created library, as the entire Python language and standard library is community created and maintained.

That said, I could update this to use [mimelib](https://pypi.org/project/mimelib/) which is an external package, and all it really does is add some gloss over the built-in package.

Reply:
Also, the narrative flow is talking about a single example and how to change it, as opposed to two at once that are subtly different. I'll reword this a bit and update to use `mimelib`.

Reply:
Ended up sticking w/ the standard `mimetypes` module because changing it would have made things even more confusing. Instead just changed the language in the text.